src/altinn-app-frontend/src/common/hooks/useFormLayoutHistory.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback, useEffect } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+
+import { useAppDispatch } from 'src/common/hooks/useAppDispatch';
+import { FormLayoutActions } from 'src/features/form/layout/formLayoutSlice';
+import type { IUpdateCurrentView } from 'src/features/form/layout/formLayoutTypes';
+
+/**
+ * @param activePageId the formLayout page id that is currently active in the app
+ * @return matchRootUrl: Matches the location pathname of the history root. If not, an empty string.
+ */
+export const useFormLayoutHistoryAndMatchInstanceLocation = ({
+  activePageId,
+}: {
+  activePageId: string;
+}): { matchRootUrl: string } => {
+  const matchRoot = useRouteMatch<string>('/instance/:partyId/:instanceGuid');
+  const matchRootUrl = matchRoot?.url || '';
+  const history = useHistory();
+  const dispatch = useAppDispatch();
+  const dispatchAction = FormLayoutActions.updateCurrentView;
+  const doDispatch = useCallback(
+    (payload: IUpdateCurrentView) => {
+      dispatch(dispatchAction(payload));
+    },
+    [dispatch, dispatchAction],
+  );
+  useEffect(() => {
+    const pageIdFromLocation = history.location.pathname.replace(
+      `${matchRootUrl}/`,
+      '',
+    );
+    const locationIsRoot =
+      !pageIdFromLocation || pageIdFromLocation === matchRootUrl;
```

Comment:
Could this be simplified with `history.location.pathname.startsWith(matchRootUrl)` in some way instead?

Reply:
rewriting.
pageIdFromLocation is either a pageId (all will be false), `''`og the `matchRootUrl`. 
So `startsWIth` would no really help.